### DISCUSSION TITLE
Jest transformer for PandaCSS

### DISCUSSION THREAD

**Discussion by isaiasmb** [chakra-ui/panda#2504] [Ideas]
Hi,
Could it be possible to have a Jest transform like [Vanilla Extract](https://vanilla-extract.style/documentation/test-environments/)?

I'm having trouble testing styles with Jest because I can't find CSS properties in the Jest environment. So I thought having a Jest transformer or something like that would be useful.

Best regards!

**Comment by astahmer** [comment]
maybe this can help ? https://github.com/chakra-ui/panda/discussions/2404#discussioncomment-8839421

**Reply by isaiasmb** [reply]
Hi @astahmer,

I'm sorry, I didn’t explain well my problem.

For example, I can't use `toHaveStyle` in my tests with Jest.

That means the CSS generated by PandaCSS is not loaded in my Jest environment.

See the example below:

![image](https://github.com/chakra-ui/panda/assets/8540555/797b21d7-fdc5-4558-a136-45473cf62403)

index.tsx
![image](https://github.com/chakra-ui/panda/assets/8540555/658b3da6-eda2-4739-8c1e-1d97eb4b6f92)

styles.ts
![image](https://github.com/chakra-ui/panda/assets/8540555/1d541f2c-64c0-4b96-a9ee-ac47bdefdd91)

Avatar.test.tsx
![image](https://github.com/chakra-ui/panda/assets/8540555/99bcb7bd-5f7b-4312-8390-6399aa91b401)

panda.config.ts
![image](https://github.com/chakra-ui/panda/assets/8540555/1524b1ad-651d-4576-8531-82bcdc4f9b9e)

jest.config.js
![image](https://github.com/chakra-ui/panda/assets/8540555/d316535b-e49e-4eac-941e-e496a5eb7268)

tests
![image](https://github.com/chakra-ui/panda/assets/8540555/7af98598-960a-484b-8929-0f43aad6433d)

Thank you!

**Reply by astahmer** [reply]
> That means the CSS generated by PandaCSS is not loaded in my Jest environment.

Panda CSS generates a static CSS file at build-time, it's up to you to import it

btw you don't need to use `cva` on the 2nd argument of `styled`, it's implicitly called

**Reply by jonambas** [reply]
IIRC, `@layer` isn't fully implemented in js dom, same with CSS variables. Try using `--polyfill` in your config or `cssgen` if you aren't already. 

I'm using Vitest, and had to use happy-dom for CSS variable support + polyfilled layers.

That said, it's probably better to test styles E2E in Playwright/Cypress so you can fully test things like media queries.
